### DISCUSSION TITLE
Wi-fi component Link to Chat editable in-game

### DISCUSSION THREAD

**Discussion by mwidomski** [FakeFishGames/Barotrauma#4976] [Ideas and suggestions]
At the moment if I understand correctly, you need to enable the link to chat option for each individual wifi component in the submarine editor (editing a sub template every time) if you want to be able to use that feature in game. Meaning that if you are mid-campaign / multiplayer game and you want to send radio commands to the component or vice-versa you would need to start a new game with a custom sub, as there is no way (that I know of) to edit a sub you are currently using as part of a save game.

Regardless, would it not be better to have the Link to Chat option of the component editable in game, just like the Channel setting is currently? That way you could enable this option selectively whenever you need it as part of a given game.

**Comment by Xeavas** [comment]
It was added in private repo, but then was reverted ): . I support of wifi LTC checkbox in game.

-Edit : the reason given was griefing, perhaps some ways to prevent that? like being receive only and doesn't output to chat? Like Listen To Chat or something along the lines.

**Reply by tiba666** [reply]
i would like to have this one added to the game but more like a rare item you could buy at a station. so griefing wouldnt be a way.

**Comment by Kaillera** [comment]
Honestly I think the amount of things you can do with it for good far outweighs the cons of griefing.

**Reply by Xeavas** [reply]
I think it was mentioned on something like a network feedback loop, but that can only happen if the wifi has the ability to output into chat, so probably a "listen" checkbox will do (for voice activated stuff) as a compromise.

**Comment by mwidomski** [comment]
I saw a [pull request](https://github.com/Regalis11/Barotrauma/pull/4531) earlier today which proposed a solution wherein the presence of the option in a particular game is in the server settings so the host can toggle it on and off before the game starts, which seems like a good solution to me.

I would further add that it may be more beneficial to split LinkToChat into two variables, one which allows for transmission from chat to the component and another from the component to chat. The former could be permanently on as a function of the component (and toggle-able in game per component) and the latter could be toggled on and off as proposed before the game starts (and also per component, if it's on in the lobby settings).

As to griefing, barring any issues which would impact the stability of the game, I'd think that the existing moderation tools (banning/kicking griefers from game) would be enough in this situation to deal with things. As for repeated output into chat as a result of the component, existing methods like MinChatMessageInterval would do for this wouldn't they?

**Reply by Sviat** [reply]
Deplete player's headphone battery for each message sent using WiFi and take amount of energy in some progression with hysteresis (i.e, in Fibonacci)
